tools/editor/scene_tree_dock.cpp

```diff
@@ -1877,6 +1895,13 @@ void SceneTreeDock::_bind_methods() {
 	ObjectTypeDB::bind_method(_MD("instance"),&SceneTreeDock::instance);
 }
 
+SceneTreeDock::NodeNameCasing SceneTreeDock::_get_node_name_casing(const String& p_property) {
+	switch (EditorSettings::get_singleton()->get(p_property).operator int()) {
+		case 1: return NODE_NAME_CASING_CAMEL_CASE;
+		case 2: return NODE_NAME_CASING_SNAKE_CASE;
+	}
+	return NODE_NAME_CASING_PASCAL_CASE;
+}
```

Comment:
@bojidar-bg Your proposal is cleverer than the current implementation, but I still don't see the point of the method at all. I'd prefer we don't implement useless stuff, even if done in a clever way ;) Just switching over `EditorSettings::get_singleton()->get("scenetree_editor/new_node_name_casing");` is a thousand times simpler, that complexity is not necessary...